src/defu.ts

```diff
@@ -28,7 +24,11 @@ function _defu<T extends defuObj> (baseObj: T | any, defaults: T | any): T {
       continue
     }
 
-    if (isObject(val) && isObject(obj[key])) {
+    if (Array.isArray(val) && Array.isArray(obj[key])) {
+      obj[key] = obj[key].concat(val)
+    } else if (typeof val === 'function' && Array.isArray(obj[key])) {
+      obj[key] = val(obj[key])
```

Comment:
This might be problematic if someone wants to apply defaults and pass function as-is to the consumer (external-lib that also accept function and calls it differently)

Reply:
Do you have a use case in mind?

Reply:
Removed :)